mlir/include/mlir/IR/Operation.h

```diff
@@ -500,6 +500,12 @@ class alignas(8) Operation final
                                llvm::to_vector(getDiscardableAttrs()));
   }
 
+  /// Return all attributes that are not stored as properties.
+  DictionaryAttr getNonPropertyAttrDictionary() {
+    return getPropertiesStorage() ? getDiscardableAttrDictionary()
+                                  : getAttrDictionary();
```

Comment:
```suggestion
    return getAttrDictionary();
```

Isn't this equivalent? Isn't this API an alias to this then?

Reply:
I think it's inverted: when properties are used, this only returns the discardable attributes (`getAttrDictionary` will include properties into the dictionary in that case). I think this is equivalent to `return attrs`.

Reply:
Ah I see, somehow I thought `getAttrDictionary` would end up with `return attrs;`

Can we name this `getRawDictionaryAttrs()` and implement it as `return attrs;`?

Reply:
Done